flang/lib/Lower/Bridge.cpp

```diff
@@ -2592,8 +2586,23 @@ class FirConverter : public Fortran::lower::AbstractConverter {
     explicitIterSpace.exprBase(&e, LHS);
   }
   void analyzeExplicitSpace(const Fortran::evaluate::Assignment *assign) {
-    analyzeExplicitSpace</*LHS=*/true>(assign->lhs);
-    analyzeExplicitSpace(assign->rhs);
+    std::visit(
+        Fortran::common::visitors{
+            [&](const Fortran::evaluate::ProcedureRef &procRef) {
+              // Ensure the procRef expressions are the one being visited.
+              assert(procRef.arguments().size() == 2);
+              const auto *lhs = procRef.arguments()[0].value().UnwrapExpr();
+              const auto *rhs = procRef.arguments()[1].value().UnwrapExpr();
+              assert(lhs && rhs &&
+                     "user defined assignment arguments must be expressions");
+              analyzeExplicitSpace</*LHS=*/true>(*lhs);
+              analyzeExplicitSpace(*rhs);
+            },
+            [&](const auto &) {
+              analyzeExplicitSpace</*LHS=*/true>(assign->lhs);
+              analyzeExplicitSpace(assign->rhs);
```

Comment:
nit: Seems like the nub of these two cases is the same and could be placed in a shared lambda, e.g.,
```c++
[](const auto &lhs, const auto &rhs) {
  analyzeExplicitSpace</*LHS=*/true>(lhs);
  analyzeExplicitSpace(rhs);
}
```

Reply:
Thanks, done.